collective/googlenews/browser/sitemap.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+from collective.googlenews.interfaces import GoogleNewsSettings
+from DateTime import DateTime
+from plone import api
+from Products.Five import BrowserView
+
+
+class GoogleNewsSiteMap(BrowserView):
+    """Google News sitemap view generate an xml file you can submit to google
+    news services to have greater refresh news"""
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def _brain2news(self, brain):
+        """Transform brain into Google News data"""
+        portal = api.portal.get()
+        return {
+            'loc': brain.getURL(),
+            'name': portal.Title(),
+            'language': (
```

Comment:
`language` should be calculated ouside of the loop, probably in the `news` function:

```python
self.language = portal.language if portal.language in ('zh-cn', 'zh-tw') else portal.language[:2]

...

'language': self.language,
```